ingestion/tests/unit/test_dbt.py

```diff
@@ -0,0 +1,1392 @@
+        port="8585",
+        path="/api/v1/users/cb2a92f5-e935-4ad7-911c-654280046538",
+    ),
+)
+
+
+class DbtUnitTest(TestCase):
+    """
+    Implements the necessary methods to extract
+    dbt Unit Test
+    """
+
+    @patch("metadata.ingestion.source.database.dbt.metadata.DbtSource.test_connection")
+    def __init__(self, methodName, test_connection) -> None:
+        super().__init__(methodName)
+        test_connection.return_value = False
+        self.config = OpenMetadataWorkflowConfig.parse_obj(mock_dbt_config)
+        self.dbt_source_obj = DbtSource.create(
+            mock_dbt_config["source"],
+            self.config.workflowConfig.openMetadataServerConfig,
+        )
+
+    @patch("metadata.ingestion.source.database.dbt.metadata.DbtSource.get_dbt_owner")
+    def test_dbt_manifest_v4_v5_v6(self, get_dbt_owner):
+        get_dbt_owner.return_value = MOCK_OWNER
+        dbt_files = DbtFiles(dbt_manifest=json.loads(MOCK_SAMPLE_MANIFEST_V4_V5_V6))
+        dbt_objects = DbtObjects(
+            dbt_catalog=parse_catalog(dbt_files.dbt_catalog)
+            if dbt_files.dbt_catalog
+            else None,
+            dbt_manifest=parse_manifest(dbt_files.dbt_manifest),
+            dbt_run_results=parse_run_results(dbt_files.dbt_run_results)
+            if dbt_files.dbt_run_results
+            else None,
+        )
+        self.check_dbt_validate(dbt_files=dbt_files)
+        self.check_yield_datamodel(dbt_objects=dbt_objects)
+
+    @patch("metadata.ingestion.source.database.dbt.metadata.DbtSource.get_dbt_owner")
+    def test_dbt_manifest_v7(self, get_dbt_owner):
+        get_dbt_owner.return_value = MOCK_OWNER
+        dbt_files = DbtFiles(dbt_manifest=json.loads(MOCK_SAMPLE_MANIFEST_V7))
+        dbt_objects = DbtObjects(
+            dbt_catalog=parse_catalog(dbt_files.dbt_catalog)
+            if dbt_files.dbt_catalog
+            else None,
+            dbt_manifest=parse_manifest(dbt_files.dbt_manifest),
+            dbt_run_results=parse_run_results(dbt_files.dbt_run_results)
+            if dbt_files.dbt_run_results
+            else None,
+        )
+        self.check_dbt_validate(dbt_files=dbt_files)
+        self.check_yield_datamodel(dbt_objects=dbt_objects)
+
+    @patch("metadata.ingestion.source.database.dbt.metadata.DbtSource.get_dbt_owner")
+    def test_dbt_manifest_v8(self, get_dbt_owner):
+        get_dbt_owner.return_value = MOCK_OWNER
+        dbt_files = DbtFiles(dbt_manifest=json.loads(MOCK_SAMPLE_MANIFEST_V8))
+        dbt_objects = DbtObjects(
+            dbt_catalog=parse_catalog(dbt_files.dbt_catalog)
+            if dbt_files.dbt_catalog
+            else None,
+            dbt_manifest=parse_manifest(dbt_files.dbt_manifest),
+            dbt_run_results=parse_run_results(dbt_files.dbt_run_results)
+            if dbt_files.dbt_run_results
+            else None,
+        )
+        self.check_dbt_validate(dbt_files=dbt_files)
+        self.check_yield_datamodel(dbt_objects=dbt_objects)
```

Comment:
Can we parametrize the test with the sample mock manifest instead of duplicating the code?

Reply:
makes sense will do that